src/lib/services/CartService.ts

```diff
@@ -0,0 +1,194 @@
+import { getItem } from '@/lib/api/StoreAPI';
+import {
+  getClientCookie,
+  getServerCookie,
+  setClientCookie,
+  setServerCookie,
+} from '@/lib/services/CookieService';
+import { PublicMerchItemOption, PublicMerchItemWithPurchaseLimits } from '@/lib/types/apiResponses';
+import { ClientCartItem, CookieCartItem } from '@/lib/types/client';
+import { CookieType } from '@/lib/types/enums';
+import { OptionsType } from 'cookies-next/lib/types';
+
+// Converters / general cart methods.
+
+const clientCartToCookie = (items: ClientCartItem[]): string =>
+  JSON.stringify(
+    items.map(item => ({
+      itemUUID: item.uuid,
+      optionUUID: item.option.uuid,
+      quantity: item.quantity,
+    }))
+  );
+
+/**
+ * Given a public merch item, convert it to a cart-friendly item.
+ * @param item The public merch item
+ * @param optionUUID The UUID of the selected option for the item
+ * @param quantity The quantity of the item to add to the cart.
+ * @returns A cart-friendly object.
+ */
+const merchItemToClientCartItem = (
+  item: PublicMerchItemWithPurchaseLimits,
+  optionUUID: string,
+  quantity: number
+): ClientCartItem => {
+  const { options, ...publicItem } = item;
+  // form the ClientCartItem from the PublicMerchItem
+  const clientCartItem: Partial<ClientCartItem> = { ...publicItem, quantity };
+  clientCartItem.option = options.find(
+    (option: PublicMerchItemOption) => option.uuid === optionUUID
+  );
+
+  // check for invalid cart cookie item
+  if (!clientCartItem.option || typeof quantity !== 'number') throw new Error();
```

Comment:
You should provide an error message in case this happens for some reason, so it's easier to debug in the future

Reply:
why was this resolved